### PR TITLE
SQL-1261: Tag triggered releases

### PR DESCRIPTION
**Pull request by nbagnard**
~~This PR does not take care of updating the packages version in the toml.
I postponed making changes both on the release manual and Evergreen script until how we want to deal with this.~~

### CODE REVIEW

**Review by terakilobyte — COMMENTED** [review]
Should we also add a "test release" task and additionally enable "Trigger Versions with Git Tags" in Evergreen prior to merging this.

Everything else looks good, though I'm not super familiar with evergreen config files yet.

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Versioning
+Versions will follow the [semantic versioning](https://semver.org/) system.  
+The following guidelines will be used to determine when each version component will be updated:
+- **major**: backwards-breaking changes
+- **minor**: functionality added in a backwards compatible manner
+- **patch**: backwards compatible bug fixes
+
+# Release Process
```
[the remark this answers is not in the document]
Comment (nbagnard) [reply RELEASE.md:8]: Good point, will do.

**Comment by nbagnard** [comment]
> Should we also add a "test release" task and additionally enable "Trigger Versions with Git Tags" in Evergreen prior to merging this.

The idea is that this is just a Github Tag, the waterfall should be green when tagging so there is not reason to test again.
I added a run of the unit test which checks the driver version, because we need to update this, and even though it is not a code change, we use the odbc package version to provide the driver version from SQLGetInfo.
I also use a cargo tool to update the packages version before compiling the release instead of relying on "sed" to hopefully avoid problems.
This should be enough, with the final manual testing in PowerBi to make sure that updating the version number did not mess up everythin.

**Review by terakilobyte — APPROVED** [review]
lgtm

**Review by pmeredit — APPROVED** [review]
LGTM

**Review by bucaojit — APPROVED** [review]
LGTM!  Thanks for adding the snapshot task.

**Comment by nbagnard** [comment]
evergreen merge

**Pull request merged**